With -Werror=implicit-function-declaration prototypes must exist.

mod_wrap doesn't need -lnsl (it's dependency of libwrap on some systems).

--- proftpd-1.3.7a/configure.in.orig	2020-08-30 11:46:50.771847975 +0200
+++ proftpd-1.3.7a/configure.in	2020-08-30 12:26:24.125657081 +0200
@@ -1367,7 +1367,10 @@
 
 dnl Checks for libraries.  Yes, this is the hard way, but it's necessary.
 AC_CACHE_CHECK(for standalone crypt,pr_cv_lib_standalone_crypt,
-  AC_TRY_LINK(,[crypt();],
+  AC_TRY_LINK([
+	       #define _MISC_SOURCE
+	       #include <unistd.h>
+	      ],[crypt("", "");],
   	pr_cv_lib_standalone_crypt="yes", pr_cv_lib_standalone_crypt="no" ))
 
 if test "$pr_cv_lib_standalone_crypt" = "no"; then
@@ -1375,7 +1378,7 @@
 fi
 
 AC_CACHE_CHECK(for standalone gethostbyname,pr_cv_lib_standalone_gethost,
-  AC_TRY_LINK(,[gethostbyname();],
+  AC_TRY_LINK([#include <netdb.h>],[gethostbyname("");],
   	pr_cv_lib_standalone_gethost="yes",
 	pr_cv_lib_standalone_gethost="no" ))
 
@@ -1385,7 +1388,12 @@
 fi
 
 AC_CACHE_CHECK(for standalone inet_aton,pr_cv_lib_standalone_aton,
-  AC_TRY_LINK(,[inet_aton();],
+  AC_TRY_LINK([
+	       #define _DEFAULT_SOURCE
+	       #include <sys/socket.h>
+	       #include <netinet/in.h>
+	       #include <arpa/inet.h>
+	      ],[inet_aton("", (void*)0);],
   	pr_cv_lib_standalone_aton="yes",
 	pr_cv_lib_standalone_aton="no" ))
 
@@ -1394,7 +1402,7 @@
 fi
 
 AC_CACHE_CHECK(for standalone nsl functions,pr_cv_lib_standalone_nsl,[
-  AC_TRY_LINK(,[gethostent();],
+  AC_TRY_LINK([#include <netdb.h>],[gethostent();],
   pr_cv_lib_standalone_nsl="yes", pr_cv_lib_standalone_nsl="no") ])
 
 if test "$pr_cv_lib_standalone_nsl" = "no"; then
@@ -1402,7 +1410,7 @@
 fi
 
 AC_CACHE_CHECK(for standalone socket functions,pr_cv_lib_standalone_sockets,
-  AC_TRY_LINK(,[bind();],
+  AC_TRY_LINK([#include <sys/socket.h>],[bind(0, (void*)0, 0);],
   pr_cv_lib_standalone_sockets="yes", pr_cv_lib_standalone_sockets="no"))
 
 if test "$pr_cv_lib_standalone_sockets" = "no"; then
--- proftpd-1.3.7a/contrib/mod_wrap.c.orig	2020-08-30 12:55:56.649387829 +0200
+++ proftpd-1.3.7a/contrib/mod_wrap.c	2020-08-30 12:56:04.492678672 +0200
@@ -22,7 +22,7 @@
  * the source code for OpenSSL in the source distribution.
  *
  * -- DO NOT MODIFY THE TWO LINES BELOW --
- * $Libraries: -lwrap -lnsl$
+ * $Libraries: -lwrap$
  */
 
 #define MOD_WRAP_VERSION "mod_wrap/1.2.4"
